src/validators/arguments.rs

```diff
@@ -117,17 +141,31 @@ impl BuildValidator for ArgumentsValidator {
             });
         }
 
+        let py_var_kwargs_mode: Bound<PyString> = match schema.get_as(intern!(py, "var_kwargs_mode"))? {
+            Some(v) => v,
+            None => PyString::new_bound(py, "single"),
+        };
+        let var_kwargs_mode = VarKwargsMode::from_str(py_var_kwargs_mode.to_string().as_str())?;
+        let var_kwargs_validator = match schema.get_item(intern!(py, "var_kwargs_schema"))? {
+            Some(v) => Some(Box::new(build_validator(&v, config, definitions)?)),
+            None => None,
+        };
+
+        if var_kwargs_mode == VarKwargsMode::UnpackedTypedDict && var_kwargs_validator.is_none() {
```

Comment:
I'm wondering if we should also check that `var_kwargs_validator` is a `TypedDictValidator`?

Reply:
Making it a `TypedDictValdiator` statically (at build time) would allow for efficiency gains (by avoiding the dispatch via `CombinedValidator`). 

But I wonder, are there cases where it can be wrapped in a function-after validator (e.g. `model_validator`)?

Reply:
> Making it a `TypedDictValdiator` statically (at build time) would allow for efficiency gains (by avoiding the dispatch via `CombinedValidator`).

The thing is (as described [here](https://github.com/pydantic/pydantic-core/pull/1451/files#r1763215151)) `var_kwargs_validator` is used for both `uniform` and `unpacked-typed-dict` modes.

> But I wonder, are there cases where it can be wrapped in a function-after validator (e.g. `model_validator`)?

only config can be attached to typed dicts iirc, and it still results in a typed dict schema.

Reply:
You could have 

```rust
#[derive(Debug, PartialEq)]
enum VarKwargsMode {
    Uniform(CombinedValidator),
    UnpackedTypedDict(TypedDictValidator),
}
```

... and replace the `FromStr` implementation with a bespoke function like `from_string_and_validator` or similar.

Reply:
Will leave this as an optimization for later, going to approve for now 👍

Reply:
See https://github.com/pydantic/pydantic-core/issues/1457